src/core/api/index.js

```diff
@@ -62,27 +63,35 @@ export function callApi({
 
   return fetch(apiURL, options)
     .then((response) => {
-      return response.clone().json()
-        .then(
-          (jsonResponse) => ({ response, jsonResponse }),
-          (error) => {
+      function makeTextResponseJson(textResponse) {
+        return { jsonResponse: { text: textResponse }, response };
+      }
+
+      if (response.headers.get('Content-Type') === 'application/json') {
```

Comment:
for extreme paranoia, how about `response.headers.get('Content-Type').toLowerCase()` ?